### PR TITLE
[test] Add configuration for Pilatus and VASP check

### PR DESCRIPTION
**Pull request by lucamar**
I propose a configuration of ReFrame on Pilatus with the Hierarchical module naming scheme, introducing the environments `cpeAMD`, `cpeCray`, `cpeGNU` and `cpeIntel` that correspond to the toolchain modules to be loaded before the apps. 
I also provide a modified check for VASP that works on Pilatus: we might use this configuration temporarily.

### CODE REVIEW

File: config/cscs.py [hunk 1]
```diff
@@ -924,12 +943,48 @@
...
+            ]
+        },
+        {
+            'name': 'cpeAMD',
             'target_systems': [
                 'eiger', 'pilatus'
             ],
             'modules': [
-                {'name': 'PrgEnv-aocc', 'collection': True}
+                {'name': 'cpeAMD', 'collection': False}
```
Comment (vkarak) [inline config/cscs.py:960]: When a module is not a collection, just use the classical way of defining it: `'modules': ['cpeAMD']`.

File: config/cscs.py [hunk 2]
```diff
@@ -660,10 +660,10 @@
             ]
         },
         {
-            'name': 'pilatus',
-            'descr': 'Alps Cray EX Supercomputer (TDS)',
+            'name': 'eiger',
```
Comment (vkarak) [inline config/cscs.py:663]: I don't understand this. Pilatus becomes Eiger and Eiger Pilatus?

File: cscs-checks/apps/vasp/vasp_check.py [hunk 3]
```diff
@@ -9,7 +9,10 @@
 
 class VASPCheck(rfm.RunOnlyRegressionTest):
     def __init__(self):
-        self.valid_prog_environs = ['builtin']
+        if self.current_system.name == 'pilatus':
+            self.valid_prog_environs = ['cpeIntel']
+        else:
+            self.valid_prog_environs = ['builtin']
```
Comment (vkarak) [inline cscs-checks/apps/vasp/vasp_check.py:15]: Add a new line after here.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: config/cscs.py (hunk 1 above)
Comment (lucamar) [reply config/cscs.py:960]: I did that, but I got an error so I changed it... maybe the error was due to something else, I can revert the change.

File: config/cscs.py (hunk 2 above)
Comment (lucamar) [reply config/cscs.py:663]: I just moved Pilatus after Eiger, since we have also Dom after Piz Daint.

File: config/cscs.py (hunk 1 above)
Comment (lucamar) [reply config/cscs.py]: I realised that initially I had specified `{'name': 'cpeAMD'}` without the `collection`, therefore I got the error.

**Review by vkarak — APPROVED** [review]

**Pull request merged**